Source/WTF/wtf/MetaAllocator.cpp

```diff
@@ -488,6 +492,15 @@ void MetaAllocator::freeFreeSpaceNode(CheckedPtr<FreeSpaceNode>&& node)
     m_mallocBalance--;
 #endif
     SUPPRESS_UNCHECKED_LOCAL auto* nodePtr = std::exchange(node, nullptr).unsafeGet(); // NOLINT
+    nodePtr->setDidBeginCheckedPtrDeletion(); // CheckedPtr handling.
+    nodePtr->~FreeSpaceNode();
+
+    // CheckedPtr handling.
+    if (nodePtr->checkedPtrCountWithoutThreadCheck()) [[unlikely]] {
+        zeroBytes(*nodePtr);
+        return;
+    }
```

Comment:
I believe you can give MetaAlloatorNode an `operator delete(void*)` that calls MetaAllocatorFreeSpaceMalloc::free. Then `delete nodePtr` here will work, which is better encapsulation of CheckedPtr.